catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/query/handlers/CqlTransformHandler.java

```diff
@@ -290,7 +314,41 @@ public Object handle(Request request, Response response) throws Exception {
       arguments = cswTransformArgumentsAdapter();
     }
 
-    attachFileToResponse(request, response, queryResponseTransformer, combinedResponse, arguments);
+    String warning = null;
+    if (resultsNotToExport.size() > 0) {
+      List<String> requiredAttr =
+          (List<String>) queryResponseTransformer.getProperty(REQUIRED_ATTR);
+      String idName = Metacard.ID;
+      if (arguments.get("columnAliasMap") != null) {
+        requiredAttr = new ArrayList<>();
+        Map<String, Serializable> columnAliasMap =
+            (Map<String, Serializable>) arguments.get("columnAliasMap");
+        idName = columnAliasMap.get(idName) == null ? idName : (String) columnAliasMap.get(idName);
+        for (String attribute :
+            (List<String>) queryResponseTransformer.getProperty(REQUIRED_ATTR)) {
+          if (StringUtils.isNotBlank((String) columnAliasMap.get(attribute))) {
+            requiredAttr.add((String) columnAliasMap.get(attribute));
+          } else {
+            requiredAttr.add(attribute);
+          }
+        }
+      }
+      if (results.size() == 0) {
+        LOGGER.debug("0 Results to export due to missing required field(s): {}", requiredAttr);
+        response.status(HttpStatus.BAD_REQUEST_400);
+        return ImmutableMap.of(
+            "message", String.format("Result(s) missing required field(s): %s", requiredAttr));
+      }
+      warning =
+          String.format("Following not exported, missing required field(s): %s", requiredAttr);
+      int count = 0;
+      for (String resultId : resultsNotToExport) {
+        warning += String.format("\\n%s) %s: %s", ++count, idName, resultId);
+      }
+    }
```

Comment:
✏️ There's a lot going on here, any way you can think of to simplify/put some of this into methods would be nice.